Floofbot/Modules/ModMail.cs

```diff
@@ -0,0 +1,195 @@
+﻿using Discord;
+using Discord.Addons.Interactive;
+using Discord.Commands;
+using Discord.WebSocket;
+using Floofbot.Services.Repository;
+using Floofbot.Services.Repository.Models;
+using Serilog;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection.Metadata;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Floofbot.Modules
+{
+    [Summary("Send a message directly to the server's moderators")]
+    [Group("modmail")]
+    public class ModMailModule : InteractiveBase
+    {
+        private FloofDataContext _floofDb;
+        static readonly ulong RFURRY_SERVER_ID =225980129799700481; // TODO: Replace this so that it works on more servers
+        public ModMailModule(FloofDataContext _floofDB)
+        {
+            _floofDb = _floofDB;
+        }
+
+        [Command("")]
+        public async Task sendModMail([Summary("Message Content")][Remainder] string content)
+        {
+            try
+            {
+                // get values
+                var serverConfig = _floofDb.ModMails.Find(RFURRY_SERVER_ID);
+                IGuild guild = Context.Client.GetGuild(RFURRY_SERVER_ID); // can return null
+                Discord.ITextChannel channel = await guild.GetTextChannelAsync((ulong)serverConfig.ChannelId); // can return null
+                IRole role = null;
+                if (serverConfig.ModRoleId != null)
+                {
+                    role = guild.GetRole((ulong)serverConfig.ModRoleId); // can return null
```

Comment:
The ulong cast is required because the database is stored as a nullable ulong (ulong?) and when using the GetRole function etc they cannot implicitly case ulong? to ulong. A get around is to change the DB so instead of using a nullable ulong it sets the value to 0. Mod role is optional and channel can be empty if it isnt configured. I don't think either solution matters though.

I moved the early return checking

Reply:
Ah, alright. I was thinking the cast should be unnecessary due to the check at L57, but if `GetRole` is picky about it, then it is what it is